tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,299 @@
+multi_test "Setup a room, and have the first user leave (SPEC-216)",
+
+    requires => [qw(
+        make_test_room change_room_powerlevels do_request_json_for user
+        more_users can_create_room
+    )],
+
+    provides => [qw( departed_room_id )],
+
+    # User A creates a room.
+    # User A invites User B to the room.
+    # User B joins the room.
+    # User B will set the ("m.room.name", "") state of the room to {
+    #   "body": "N1. B's room name before A left"
+    # }
+    # User B will set the ("madeup.test.state", "") state of the room to {
+    #   "body": "S1. B's state before A left"
+    # }
+    # User B will send a message with body "M1. B's message before A left"
+    # User B will send a message with body "M2. B's message before A left"
+    # User A will leave the room.
+    # User B will set the ("m.room.name", "") state of the room to {
+    #   "body": "N2. B's room name after A left"
+    # }
+    # User B will set the ("madeup.test.state", "") state of the room to {
+    #   "body": "S2. B's state after A left"
+    # }
+    # User B will send a message with text "M3. B's message after A left"
+    #
+    do => sub {
+        my (
+            $make_test_room, $change_room_powerlevels, $do_request_json_for,
+            $user_a, $more_users
+        ) = @_;
+        my $user_b = $more_users->[1];
+
+        my $room_id;
```

Comment:
In fact if you just pull this out to the toplevel in the file, the other `test` blocks will be able to see it.